rest_framework/fields.py

```diff
@@ -59,7 +59,11 @@ def is_simple_callable(obj):
     if not (function or method):
         return False
 
-    args, _, _, defaults = inspect.getargspec(obj)
+    if six.PY2:
+        args, _, _, defaults = inspect.getargspec(obj)
+    else:
+        args, _, _, defaults = inspect.getfullargspec(obj)[:4]
+
```

Comment:
@rpkilby `compat.getargspec` only returns the  method `parameters` and `defaults` and that's all `fields.is_simple_callable` cares about and nothing else.

If other parts of the application requires something more than that then from `compat.getargspec` then modifications can be applied, otherwise the whole backward compatibility should be taken care of in `fields.is_simple_callable` itself, which that was what I did in the first place and @tomchristie suggested the change, In my opinion it keeps the code clean and straight.